repository_service_tuf_api/__init__.py

```diff
@@ -97,60 +100,70 @@ def sync_redis():
         redis_loader.write(settings_repository, settings_repository.to_dict())
 
 
-# Tokens
-if secrets_settings.TOKEN_KEY.startswith("/run/secrets/"):
-    try:
-        with open(secrets_settings.TOKEN_KEY) as f:
-            SECRET_KEY = f.read().rstrip("\n")
-    except OSError as err:
-        logging.error(str(err))
-
-else:
-    SECRET_KEY = secrets_settings.TOKEN_KEY
-
-if secrets_settings.ADMIN_PASSWORD.startswith("/run/secrets/"):
-    try:
-        with open(secrets_settings.ADMIN_PASSWORD) as f:
-            ADMIN_PASSWORD = f.read().rstrip("\n")
-    except OSError as err:
-        logging.error(str(err))
+SECRET_KEY: Optional[str] = None
+ADMIN_PASSWORD: Optional[str] = None
+db: Optional[sessionmaker] = None
 
-else:
-    ADMIN_PASSWORD = secrets_settings.ADMIN_PASSWORD
+if is_auth_enabled is True:
+    # Tokens
+    if secrets_settings.TOKEN_KEY.startswith("/run/secrets/"):
+        try:
+            with open(secrets_settings.TOKEN_KEY) as f:
+                SECRET_KEY = f.read().rstrip("\n")
+        except OSError as err:
+            logging.error(str(err))
 
-# User database setup
-DATABASE_URL = settings.get(
-    "DATABASE_URL", f"sqlite:///{os.path.join(DATA_DIR, 'users.sqlite')}"
-)
-
-engine = create_engine(DATABASE_URL, connect_args={"check_same_thread": False})
-SessionLocal = sessionmaker(autocommit=False, autoflush=False, bind=engine)
+    else:
+        SECRET_KEY = secrets_settings.TOKEN_KEY
 
+    if secrets_settings.ADMIN_PASSWORD.startswith("/run/secrets/"):
+        try:
+            with open(secrets_settings.ADMIN_PASSWORD) as f:
+                ADMIN_PASSWORD = f.read().rstrip("\n")
+        except OSError as err:
+            logging.error(str(err))
```

Comment:
Same here we can add a simple test for that case as well.

Reply:
For now, we won't fix this.
See: https://github.com/repository-service-tuf/repository-service-tuf-api/pull/329#discussion_r1182549739